rest/config.go

```diff
@@ -1289,6 +1289,9 @@ func SetupServerContext(ctx context.Context, config *StartupConfig, persistentCo
 // fetchAndLoadConfigs retrieves all database configs from the ServerContext's bootstrapConnection, and loads them into the ServerContext.
 // It will remove any databases currently running that are not found in the bucket.
 func (sc *ServerContext) fetchAndLoadConfigs(ctx context.Context, isInitialStartup bool) (count int, err error) {
+	sc.lock.Lock()
+	defer sc.lock.Unlock()
```

Comment:
I'm not sure this addition looks right - we're trying to get a read lock and/or a write lock later on in this function which will deadlock.

The 3.0.1 version of `fetchAndLoadConfigs` looks significantly different and does all of its locking like this in the entire function (which blocked on loading configs over the network). Since then it's changed to be more granular and only to be when applying config.